lib/kube/proxy/self_subject_reviews.go

```diff
@@ -334,11 +342,10 @@ func (m *kubernetesResourceMatcher) Match(role types.Role, condition types.RoleC
 				return ok, trace.Wrap(err)
 			}
 		} else {
-			if ok, err := utils.SliceMatchesRegex(namespace, []string{resource.Namespace}); err != nil || ok || namespace == "" {
```

Comment:
why did we remove the empty namespace here?

Reply:
Beacsue empty namespace means the user made a request with --all-namespaces, we don't want to match just on that.

Reply:
Would you have an example of scenario that would fit this?